src/radical/pilot/raptor/master.py

```diff
@@ -46,29 +53,60 @@ def __init__(self, cfg=None, backend='zmq'):
 
         rpu.Component.__init__(self, cfg, self._session)
 
-        self.register_output(rps.AGENT_STAGING_INPUT_PENDING,
-                             rpc.AGENT_STAGING_INPUT_QUEUE)
-
         self.register_publisher(rpc.STATE_PUBSUB)
         self.register_publisher(rpc.CONTROL_PUBSUB)
 
         self.register_subscriber(rpc.STATE_PUBSUB,   self._state_cb)
         self.register_subscriber(rpc.CONTROL_PUBSUB, self._control_cb)
 
-        # set up RU ZMQ Queues for request distribution and result collection
-        req_cfg = ru.Config(cfg={'channel'    : '%s.to_req' % self._uid,
-                                 'type'       : 'queue',
-                                 'uid'        : self._uid + '.req',
-                                 'path'       : os.getcwd(),
-                                 'stall_hwm'  : 0,
-                                 'bulk_size'  : 56})
-
-        res_cfg = ru.Config(cfg={'channel'    : '%s.to_res' % self._uid,
-                                 'type'       : 'queue',
-                                 'uid'        : self._uid + '.res',
-                                 'path'       : os.getcwd(),
-                                 'stall_hwm'  : 0,
-                                 'bulk_size'  : 56})
+        # send new worker tasks and agent input staging / agent scheduler
+        self.register_output(rps.AGENT_STAGING_INPUT_PENDING,
+                             rpc.AGENT_STAGING_INPUT_QUEUE)
+
+        # set up zmq queues between the agent scheduler and this master so that
+        # we can receive new requests from RP tasks
+        qname = '%s_input_queue' % self._uid
+        input_cfg = ru.Config(cfg={'channel'   : qname,
+                                   'type'      : 'queue',
+                                   'uid'       : '%s_input' % self._uid,
+                                   'path'      : os.getcwd(),
+                                   'stall_hwm' : 0,
+                                   'bulk_size' : 56})
+
+        self._input_queue  = ru.zmq.Queue(input_cfg)
+        self._input_queue.start()
+
+        # begin to receive tasks in that queue
+        self._input_getter = ru.zmq.Getter(qname,
+                                      self._input_queue.addr_get,
+                                      cb=self._request_cb)
+
+        # and register that input queue with the scheduler
+        self.publish(rpc.CONTROL_PUBSUB,
+                      {'cmd': 'register_raptor_queue',
+                       'arg': {'name' : self._uid,
+                               'queue': qname,
+                               'addr' : str(self._input_queue.addr_put)}})
```

Comment:
this part should be moved to the end of initialization method, after `self._info = {'req_addr_get':...`, otherwise it immediately starts getting data, but not all attributes are initialized and it causes errors

Reply:
part L66-L89